src/librustc_typeck/coherence/mod.rs

```diff
@@ -268,29 +268,37 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
 
             match param_env.can_type_implement_copy(tcx, self_type, span) {
                 Ok(()) => {}
-                Err(CopyImplementationError::InfrigingField(name)) => {
-                    struct_span_err!(tcx.sess,
-                                     span,
-                                     E0204,
-                                     "the trait `Copy` may not be implemented for this type")
-                        .span_label(span, &format!("field `{}` does not implement `Copy`", name))
-                        .emit()
+                Err(CopyImplementationError::InfrigingField(did, name)) => {
+                    let mut err = struct_span_err!(tcx.sess,
+                                                   span,
+                                                   E0204,
+                                                   "the trait `Copy` may not be implemented for \
+                                                   this type");
+                    err.span_label(span, &format!("field `{}` does not implement `Copy`", name));
+                    if let Some(def_span) = tcx.map.span_if_local(did) {
+                        err.span_label(def_span, &format!("`{}` defined here", name));
```

Comment:
I don't see tests for this "defined here" branch. There's a test for variant "defined here" case, but not for fields. Can you add one if there aren't?